src/Language/PureScript/Sugar.hs

```diff
@@ -68,4 +68,5 @@ desugar = map removeSignedLiterals
           >=> desugarImports
           >=> rebracket
           >=> desugarTypeClasses
+          >=> desugarCasesModule -- TODO: avoid duplicating this
```

Comment:
No it doesn't, it's just that desugaring typeclasses reintroduces more cases. I tried juggling the order but my attempts resulted in other problems.